### PR TITLE
Adding spotless to gradle

### PR DESCRIPTION
**Pull request by Kjames5269**
Updating the Git Ignore
Adding Spotless
Adding License

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+*.DS_Store
+.history
+jsconfig.json
+.vscode/
+node_modules/
+node/
+atlassian-ide-plugin.xml
+*.sonar-ide.properties
+**/webapp/css/styles.css
+platform/solr/**/overlays
```
Comment (emmberk) [inline .gitignore]: ✏️ Are some these lines from DDF? Are they necessary?
- *.checkstyle
- *.DS_Store
- .history
- jsconfig.json
- node_modules/
- node/
- atlassian-ide-plugin.xml
- *.sonar-ide.properties
- **/webapp/css/styles.css
- platform/solr/**/overlays

File: build.gradle [hunk 2]
```diff
@@ -1,6 +1,54 @@
+plugins {
+    id 'java'
+    id 'maven-publish'
+    id 'com.palantir.docker' version '0.21.0'
+    id 'com.palantir.docker-compose' version '0.21.0'
```
Comment (emmberk) [inline build.gradle]: ✏️ Is this dicker stuff related to your PR?

File: license.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3
+of the License, or any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details. A copy of the
+GNU Lesser General Public License is distributed along with this
+program and can be found at http://www.gnu.org/licenses/lgpl.html.
```
Comment (emmberk) [inline license.java]: Is this the right comment formatting for a java header? I would have expected it to look like
```java
/*
 * ...
 */
```

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,6 @@
+#Fri Apr 26 13:57:56 MST 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-all.zip
```
Comment (emmberk) [inline gradle/wrapper/gradle-wrapper.properties]: ❓What is the difference between `gradle-4.10.3-bin` and `gradle-4.10.3-all`?

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -1,5 +1,6 @@
+#Fri Apr 26 13:57:56 MST 2019
```
Comment (emmberk) [inline gradle/wrapper/gradle-wrapper.properties]: ✏️ Remove this line

File: build.gradle [hunk 6]
```diff
@@ -1,6 +1,54 @@
...
+        java {
+            licenseHeaderFile rootProject.file("license.java")
+            removeUnusedImports()
+            googleJavaFormat()
+            targetExclude '**/GatewayApplication.java'
+        }
+    }
+}
+
+subprojects {
```
Comment (emmberk) [inline build.gradle]: ❓ Do we have any subprojects?

File: build.gradle [hunk 7]
```diff
@@ -1,6 +1,54 @@
...
+
+    sourceCompatibility = 1.11
+    targetCompatibility = 1.11
+
+    spotless {
+        java {
+            licenseHeaderFile rootProject.file("license.java")
+            removeUnusedImports()
+            googleJavaFormat()
+            targetExclude '**/GatewayApplication.java'
```
Comment (emmberk) [inline build.gradle]: ❓What does this line do?

File: build.gradle [hunk 8]
```diff
@@ -1,6 +1,54 @@
...
+subprojects {
+
+    buildscript {
+        repositories {
+            mavenLocal()
+            mavenCentral()
+        }
+
+        dependencies {
+            classpath 'org.springframework.boot:spring-boot-gradle-plugin:2.1.3.RELEASE'
```
Comment (emmberk) [inline build.gradle]: ✏️ Is this change related to your PR?

File: build.gradle [hunk 9]
```diff
@@ -1,6 +1,54 @@
...
+allprojects {
+
+    apply plugin: 'com.diffplug.gradle.spotless'
+    apply plugin: 'maven-publish'
+    apply plugin: 'java'
+
+    sourceCompatibility = 1.11
+    targetCompatibility = 1.11
+
+    spotless {
```
Comment (emmberk) [inline build.gradle]: ❗️Add documentation to the README.md about `gradlew spotlessApply`

File: build.gradle [hunk 10]
```diff
@@ -1,6 +1,54 @@
...
+
+        dependencies {
+            classpath 'org.springframework.boot:spring-boot-gradle-plugin:2.1.3.RELEASE'
+        }
+    }
+
+    repositories {
+        mavenLocal()
+        maven { url "https://repo.spring.io/libs-snapshot" }
+        maven { url "https://repo.spring.io/libs-milestone" }
```
Comment (emmberk) [inline build.gradle]: ✏️ Are these 2 spring.io lines related to your PR?

File: build.gradle [hunk 11]
```diff
@@ -1,6 +1,54 @@
...
\ No newline at end of file
+}
+
+allprojects {
+
+    apply plugin: 'com.diffplug.gradle.spotless'
+    apply plugin: 'maven-publish'
+    apply plugin: 'java'
+
+    sourceCompatibility = 1.11
```
Comment (jhunzik) [inline build.gradle]: ❓ Is it supposed to be `1.11` or `'11'`?

File: build.gradle [hunk 12]
```diff
@@ -1,6 +1,54 @@
...
+            mavenLocal()
+            mavenCentral()
+        }
+
+        dependencies {
+            classpath 'org.springframework.boot:spring-boot-gradle-plugin:2.1.3.RELEASE'
+        }
+    }
+
+    repositories {
```
Comment (jhunzik) [inline build.gradle]: ✏️ Do we need this `repositories` and the one in `buildscript` too?

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (jhunzik) [reply gradle/wrapper/gradle-wrapper.properties]: Follow-on question: Do we have any reason to not use the latest version: Gradle 5.x.x?

File: .gitignore (hunk 1 above)
Comment (Kjames5269) [reply .gitignore]: Good question, probably not 🤷‍♂ I'll remove them and only add in the stuff that's needed for this PR

File: build.gradle (hunk 2 above)
Comment (Kjames5269) [reply build.gradle]: Not this one, but it'l be needed as soon as we implement docker. I'll wait on it

File: license.java (hunk 3 above)
Comment (Kjames5269) [reply license.java]: 🤦‍♂ nice catch

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (Kjames5269) [reply gradle/wrapper/gradle-wrapper.properties]: Both are questions I don't know the answer to. We should switch to 5.x.x before the gradle file becomes tech debt

File: build.gradle (hunk 11 above)
Comment (jhunzik) [reply build.gradle]: Josh, it should be the first one.

Comment (Kjames5269) [reply build.gradle]: 1.11

File: build.gradle (hunk 6 above)
Comment (Kjames5269) [reply build.gradle]: No idea but I copied from SamlConf. I'll put it in allProjects for now

File: build.gradle (hunk 7 above)
Comment (Kjames5269) [reply build.gradle]: Excludes a file called GatewayApplication from the style checks. 

That file doesn't exist anymore.

...

File: build.gradle (hunk 8 above)
Comment (Kjames5269) [reply build.gradle]: No but It'll be needed once we start using spring boot. I'll hold off on it

File: build.gradle (hunk 12 above)
Comment (Kjames5269) [reply build.gradle]: No idea got rid of it for now

File: build.gradle (hunk 9 above)
Comment (Kjames5269) [reply build.gradle]: :+1:

File: gradlew [hunk 13]
```diff
@@ -28,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (jhunzik) [inline gradlew]: Is there any harm in leaving this empty?

Comment (Kjames5269) [reply gradlew]: It's auto generated 🤷‍♂ I'd guess that means it runs the jvm with 64MB of memory

File: build.gradle [hunk 14]
```diff
@@ -1,6 +1,38 @@
+plugins {
+    id 'java'
+    id 'maven-publish'
```
Comment (jhunzik) [inline build.gradle]: Why do we need this maven plugin?

Comment (Kjames5269) [reply build.gradle]: For releases I think

Comment (Kjames5269) [reply build.gradle]: But I'll remove it til we need it

**Comment by Kjames5269** [comment]
build now

**Comment by Kjames5269** [comment]
build now
(Ran spotless Apply)
